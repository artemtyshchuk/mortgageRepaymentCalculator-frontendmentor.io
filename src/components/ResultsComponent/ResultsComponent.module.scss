.resultsComponent {
}
.resultSideTitle {
  color: var(--color-white, #fff);
  /* desktop/text-preset-2 */
  font-family: "Plus Jakarta Sans";
  font-size: var(--font-size-text-preset-2, 24px);
  font-style: normal;
  font-weight: 700;
  line-height: 125%; /* 30px */
}

.resultSideTitle {
  color: var(--color-white, #fff);

  /* desktop/text-preset-2 */
  font-family: "Plus Jakarta Sans";
  font-size: var(--font-size-text-preset-2, 24px);
  font-style: normal;
  font-weight: 700;
  line-height: 125%; /* 30px */
}

.resultSideDescription {
  color: var(--color-slate-300, #9abed5);

  /* desktop/text-preset-4 */
  font-family: "Plus Jakarta Sans";
  font-size: var(--font-size-text-preset-4, 16px);
  font-style: normal;
  font-weight: 500;
  line-height: 150%; /* 24px */
  align-self: stretch;
}

.resultSideResultsContainer {
  width: calc(100% - 64px);
  padding: 32px;
  border-radius: var(--spacing-100, 8px);
  border-top: 4px solid var(--color-lime, #d8db2f);
  background: rgba(0, 0, 0, 0.25);
}

.repaymentsText {
  color: var(--color-slate-300, #9abed5);
  margin: 0;
  /* desktop/text-preset-4 */
  font-family: "Plus Jakarta Sans";
  font-size: var(--font-size-text-preset-4, 16px);
  font-style: normal;
  font-weight: 500;
  line-height: 150%; /* 24px */
}

.repaymentsInfo {
  color: var(--color-lime, #d8db2f);
  margin: 0;
  padding-top: 8px;
  /* desktop/text-preset-1 */
  font-family: "Plus Jakarta Sans";
  font-size: var(--font-size-text-preset-1, 56px);
  font-style: normal;
  font-weight: 700;
  line-height: 125%; /* 70px */
}

.divider {
  display: flex;
  height: 1px;
  width: 100%;
  margin: 32px 0;
  background: rgba(154, 190, 213, 0.25);
}

.totalRepaymentsInfo {
  color: var(--color-white, #fff);
  margin: 0;
  padding-top: 8px;
  /* desktop/text-preset-2 */
  font-family: "Plus Jakarta Sans";
  font-size: var(--font-size-text-preset-2, 24px);
  font-style: normal;
  font-weight: 700;
  line-height: 125%; /* 30px */
}

@media (max-width: 1439px) {
  //1201
}

@media (max-width: 1200px) {
  //993
}

@media (max-width: 992px) {
  //769
}

@media (max-width: 768px) {
  //577
}

@media (max-width: 576px) {
}
