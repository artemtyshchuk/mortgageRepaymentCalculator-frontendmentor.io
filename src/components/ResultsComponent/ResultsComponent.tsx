import styles from "./ResultsComponent.module.scss";

interface ResultsComponentProps {
  monthlyRepayment: number;
  totalRepayments: number;
}

export const ResultsComponent = ({
  monthlyRepayment,
  totalRepayments,
}: ResultsComponentProps) => {
  return (
    <div className={styles.resultsComponent} data-testid="resultsComponent">
      <p className={styles.resultSideTitle}>Your results</p>
      <p className={styles.resultSideDescription}>
        Your results are shown below based on the information you provided. To
        adjust the results, edit the form and click “calculate repayments”
        again.
      </p>
      {/* resultSideResultsContainer */}
      <div className={styles.resultSideResultsContainer}>
        <p className={styles.repaymentsText}>Your monthly repayments</p>
        <p className={styles.repaymentsInfo}>£{monthlyRepayment.toFixed(2)}</p>
        <span className={styles.divider}></span>
        <p className={styles.repaymentsText}>
          Total you'll repay over the term
        </p>
        <p className={styles.totalRepaymentsInfo} data-testid="totalRepayments">
          £{totalRepayments.toFixed(2)}
        </p>
      </div>
      {/* <-----> */}
    </div>
  );
};
