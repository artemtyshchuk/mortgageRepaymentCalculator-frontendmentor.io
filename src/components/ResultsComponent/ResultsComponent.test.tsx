import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"; // для расширенных матчеров Jest
import { ResultsComponent } from "./ResultsComponent";

const testData = {
  monthlyRepayment: 123.45,
  totalRepayments: 6789.01,
};

describe("ResultsComponent", () => {
  it("renders the component with the correct data", () => {
    render(
      <ResultsComponent
        monthlyRepayment={testData.monthlyRepayment}
        totalRepayments={testData.totalRepayments}
      />
    );

    expect(screen.getByText("Your results")).toBeInTheDocument();
    expect(
      screen.getByText(
        "Your results are shown below based on the information you provided. To adjust the results, edit the form and click “calculate repayments” again."
      )
    ).toBeInTheDocument();

    expect(
      screen.getByText(`£${testData.monthlyRepayment.toFixed(2)}`)
    ).toBeInTheDocument();
    expect(screen.getByTestId("totalRepayments")).toHaveTextContent(
      `£${testData.totalRepayments.toFixed(2)}`
    );
    expect(screen.getByTestId("resultsComponent")).toMatchSnapshot();
  });
});
