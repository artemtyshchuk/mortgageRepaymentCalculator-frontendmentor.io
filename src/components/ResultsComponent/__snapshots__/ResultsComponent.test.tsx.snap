// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsComponent renders the component with the correct data 1`] = `
<div
  class="resultsComponent"
  data-testid="resultsComponent"
>
  <p
    class="resultSideTitle"
  >
    Your results
  </p>
  <p
    class="resultSideDescription"
  >
    Your results are shown below based on the information you provided. To adjust the results, edit the form and click “calculate repayments” again.
  </p>
  <div
    class="resultSideResultsContainer"
  >
    <p
      class="repaymentsText"
    >
      Your monthly repayments
    </p>
    <p
      class="repaymentsInfo"
    >
      £
      123.45
    </p>
    <span
      class="divider"
    />
    <p
      class="repaymentsText"
    >
      Total you'll repay over the term
    </p>
    <p
      class="totalRepaymentsInfo"
      data-testid="totalRepayments"
    >
      £
      6789.01
    </p>
  </div>
</div>
`;
